### PR TITLE
registry: Add support for a custom logger

### PR DESCRIPTION
**Pull request by clrprod**

### CODE REVIEW

File: pkg/registry/registry.go [hunk 1]
```diff
@@ -74,5 +78,18 @@ func New() http.Handler {
...
+		o(r)
+	}
+	return http.HandlerFunc(r.root)
+}
+
+// Options describes the available options
+// for creating the registry.
+type Option func(r *registry)
+
+func Logger(l *log.Logger) Option {
```
Comment (ImJasonH) [inline pkg/registry/registry.go]: `WithLogger`?

Comment (nhooyr) [reply pkg/registry/registry.go]: I'd prefer `Logger` as it's cleaner.

Comment (clrprod) [reply pkg/registry/registry.go]: Most options don't seem to add a verb at the front, and it makes them shorter. This also only pollutes the registry namespace which has New(), TLS() and now Logger()

**Comment by nhooyr** [comment]
@googlebot I consent.

**Pull request merged**